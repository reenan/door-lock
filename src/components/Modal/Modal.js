import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './Modal.scss'

class Modal extends Component {
	clickOnOverlay = () => {
		if (this.props.closeWithClickOnOverlay) {
			this.props.close()
		}
	}

	render() {
		const { open, children, className } = this.props

		return (
			open ?
				<div className={`${className} modal open`}>
					<div onClick={this.clickOnOverlay} className={`overlay`} />
					<div className='modal-inner'>
						<div className='modal-content-area'>
							{children}
						</div>
					</div>
				</div> : null
		)
	}
}

Modal.propTypes = {
	closeWithClickOnOverlay: PropTypes.bool,
	open: PropTypes.bool,
	close: PropTypes.func,
	className: PropTypes.string,
}

Modal.defaultProps = {
	closeWithClickOnOverlay: true,
	open: false,
	className: '',
	close: () => {},
}

export default Modal
