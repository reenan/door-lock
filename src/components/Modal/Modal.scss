@import '../../resources/_mixins.scss';

.modal {
	position: fixed;
	top: 0;
	left: 0;
	width: 100vw;
	height: 100vh;
	text-align: center;
	display: table;
	color: black;
	z-index: 5;

	pointer-events: none;
	opacity: 0;

	.overlay {
		@include absolute-fill();
		background-color: black;
		pointer-events: none;
		opacity: 0;
		z-index: 1;
	}

	&.open {
		opacity: 1;
		pointer-events: auto;

		.overlay {
			opacity: 0.5;
			pointer-events: auto;
		}
	}

	.modal-inner {
		display: table-cell;
		vertical-align: middle;

		.modal-content-area {
			@include box-shadow(0px 3px 5px 0px, rgba(0, 0, 0, 0.15));
			box-sizing: border-box;
			text-align: left;
			position: relative;
			overflow-x: hidden;
			z-index: 2;
			display: inline-block;
			background-color: white;
			padding: 20px 35px 20px 35px;
      border-radius: 5px;
      max-width: 85%;

			.button {
				margin-top: 15px;
			}
		}
	}
}