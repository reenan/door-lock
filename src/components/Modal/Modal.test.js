import React from 'react'
import Modal from './Modal'

import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const component = renderer.create(
    <Modal />
  )

  expect(component).toMatchSnapshot()
})

it('renders without crashing with open', () => {
  const component = renderer.create(
    <Modal open />
  )

  expect(component).toMatchSnapshot()
})

it('show call close prop when clicked on overlay', () => {
  const closeMock = jest.fn()

  const component = renderer.create(
    <Modal open close={closeMock} />
  )

  component.getInstance().clickOnOverlay()
  expect(closeMock).toHaveBeenCalled()
})

it('show not call close prop when clicked on overlay and closeWithClickonOverlay as false', () => {
  const closeMock = jest.fn()

  const component = renderer.create(
    <Modal open closeWithClickOnOverlay={false} close={closeMock} />
  )

  component.getInstance().clickOnOverlay()
  expect(closeMock).not.toHaveBeenCalled()
})
