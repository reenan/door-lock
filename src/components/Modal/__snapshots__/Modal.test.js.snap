// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `null`;

exports[`renders without crashing with open 1`] = `
<div
  className=" modal open"
>
  <div
    className="overlay"
    onClick={[Function]}
  />
  <div
    className="modal-inner"
  >
    <div
      className="modal-content-area"
    />
  </div>
</div>
`;
